src/CommunityToolkit.Maui/Extensions/Popup/PopupExtensions.shared.cs

```diff
@@ -0,0 +1,108 @@
+﻿using CommunityToolkit.Maui.Core;
+using CommunityToolkit.Maui.Views;
+using Microsoft.Maui.Platform;
+
+namespace Microsoft.Maui.Controls;
+
+/// <summary>
+/// Extension methods for <see cref="Popup"/>.
+/// </summary>
+public static partial class PopupExtensions
+{
+	/// <summary>
+	/// Displays a popup.
+	/// </summary>
+	/// <param name="page">
+	/// The current <see cref="Page"/>.
+	/// </param>
+	/// <param name="popup">
+	/// The <see cref="BasePopup"/> to display.
+	/// </param>
+	public static void ShowPopup<TPopup>(this Page page, TPopup popup) where TPopup : BasePopup
```

Comment:
@brminnick why not use the `IPopup` interface instead?

Reply:
In `PopupExtensions.windows.cs`, `PlatformShowPopup` sets the value of `popup.Parent`:

https://github.com/CommunityToolkit/Maui/blob/7afed071cd46498267fd1a9b4d6d8c8737f1e623/src/CommunityToolkit.Maui/Extensions/Popup/PopupExtensions.windows.cs#L13

Reply:
Oh yeah, we can't use the `IPopup`. I just looked at the old code, and we can use the `BasePopup` since all Popups will be inherited from it